docs/references/config/config.toml.md

```diff
@@ -0,0 +1,2055 @@
+instead of fetching and replaying historical blocks. It requires some peers in the network to take and serve state
+machine snapshots. State sync is not attempted if the node has any local state (LastBlockHeight > 0).
+
+The node will have a truncated block history, starting from the height of the snapshot.
+
+### statesync.enable
+Enable state synchronization.
+```toml
+enable = false
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `false` |
+|                     | `true`  |
+
+Enable state synchronization on first start.
+
+### statesync.rpc_servers
+Comma-separated list of RPC servers for light client verification of the synced state machine,
+and retrieval of state data for node bootstrapping.
+```toml
+rpc_servers = ""
+```
+
+| Value type                        | string (comma-separated list)      |
+|:----------------------------------|:-----------------------------------|
+| **Possible values within commas** | nodeID@IP:port (`"1.2.3.4:26657"`) |
+|                                   | `""`                               |
+
+At least two RPC servers have to be defined for state synchronization to work.
+
+### statesync.trust_height
+The height of the trusted header hash.
+```toml
+trust_height = 0
+```
+
+| Value type          | integer |
+|:--------------------|:--------|
+| **Possible values** | &gt;= 0 |
+
+`0` is only allowed when state synchronization is disabled.
+
+### statesync.trust_hash
+Header hash obtained from a trusted source.
+```toml
+trust_hash = ""
+```
+
+| Value type          | string             |
+|:--------------------|:-------------------|
+| **Possible values** | hex-encoded number |
+|                     | ""                 |
+
+`""` is only allowed when state synchronization is disabled.
+
+This is the header hash value obtained from the trusted source at height
+[statesync.trust_height](#statesynctrust_height).
+
+### statesync.trust_period
+The period during which validators can be trusted.
+```toml
+trust_period = "168h0m0s"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0s"`      |
+
+For Cosmos SDK-based chains, `statesync.trust_period` should usually be about 2/3rd of the unbonding period
+(about 2 weeks) during which they can be financially punished (slashed) for misbehavior.
+
+### statesync.discovery_time
+Time to spend discovering snapshots before initiating a restore.
+```toml
+discovery_time = "15s"
+```
+
+<!--- What happens when this time expires? --->
+
+### statesync.temp_dir
+Temporary directory for state sync snapshot chunks.
+```toml
+temp_dir = ""
+```
+
+| Value type          | string                  |
+|:--------------------|:------------------------|
+| **Possible values** | undefined               |
+
+This value is unused by CometBFT. It was not hooked up to the state sync reactor.
```

Comment:
I propose to word the `mempool.wal_dir` section the same as this one, while we decide what to do with the code (either remove `mempool.wal_dir` from the config, of wire it to the actual logic).

Reply:
Done.